frontend/www/js/omegaup/components/contest/FilteredList.vue

```diff
@@ -0,0 +1,156 @@
+              </td>
+              <td class="col-md-2" v-if="showPractice">
+                <a v-bind:href="`/arena/${contest.alias}/practice/`">
+                  <span>{{ T.wordsPractice }}</span>
+                </a>
+              </td>
+              <td class="col-md-2" v-if="showPractice">
+                <a v-bind:href="`/arena/${contest.alias}/#ranking`">
+                  <span>{{ T.wordsContestsResults }}</span>
+                </a>
+              </td>
+              <td class="col-md-2" v-if="!UI.isVirtual(contest) && showVirtual">
+                <a v-bind:href="`/arena/${contest.alias}/virtual/`">
+                  <span>{{ T.virtualContest }}</span>
+                </a>
+              </td>
+              <td class="no-wrap col-md-2" v-if="$parent.showPublicUpdated">
+                {{ contest.last_updated.long() }}
+              </td>
+            </tr>
+            <tr>
+              <td colspan="5" class="forcebreaks forcebreaks-arena">
+                {{ contest.description }}
+              </td>
+            </tr>
+          </tbody>
+          <tfoot>
+            <tr v-if="hasNext || hasPrevious" align="center">
+              <td class="no-wrap" v-bind:colspan="pagerColumns">
+                <a v-if="hasPrevious" v-on:click="previous" href="#">{{
+                  T.wordsPrevPage
+                }}</a>
+                <span class="page-num">{{ pageNumber }}</span>
+                <a v-if="hasNext" v-on:click="next" href="#">{{
+                  T.wordsNextPage
+                }}</a>
+              </td>
+            </tr>
+          </tfoot>
+        </table>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component, Prop, Emit } from 'vue-property-decorator';
+import { T } from '../../omegaup.js';
+import UI from '../../ui.js';
+import omegaup from '../../api.js';
+
+@Component({})
+export default class List extends Vue {
+  @Prop() contests!: omegaup.Contest[];
+  @Prop() showTimes!: boolean;
+  @Prop() showPractice!: boolean;
+  @Prop() showVirtual!: boolean;
+  @Prop() showPublicUpdated!: boolean;
+  @Prop() recommended!: boolean;
+
+  T = T;
+  UI = UI;
+  pageNumber = 1;
+  pageSize = 10;
+
+  get totalPages(): number {
+    return Math.ceil(this.contests.length / this.pageSize);
+  }
+
+  get page(): omegaup.Contest[] {
+    let first = (this.pageNumber - 1) * this.pageSize;
+    return this.contests.slice(first, first + this.pageSize);
+  }
+
+  get hasPrevious(): boolean {
+    return this.pageNumber > 1;
+  }
+
+  get hasNext(): boolean {
+    return this.pageNumber < this.totalPages;
+  }
+
+  get pagerColumns(): number {
+    let cols = 2;
+    if (this.showPractice) cols += 1;
+    if (this.showVirtual) cols += 1;
+    if (this.showTimes) cols += 3;
+    if (this.showPublicUpdated) cols += 1;
+    return cols;
+  }
+
+  next() {
+    // TODO: Update history so the back button works correctly.
+    if (this.pageNumber < this.totalPages) {
```

Comment:
nit: usa el guard clause pattern para que sea igual que L145.

```suggestion
    if (this.pageNumber >= this.totalPages) {
      return;
    }
```

Reply:
Listo